### PR TITLE
Update Manipulations.php to let manipulations arrays also

### PR DESCRIPTION
**Pull request by Krato**
WIth this fix you can send a normal manipulation:

```php
$media->manipulations = [
    'resized' =>['width' => 123, 'height' => 123],
];
```

And also this kind of manipulations:

```php
$resizedConversion = Manipulations::create()
    ->width(123)
    ->apply()
    ->manualCrop(20, 20, 10, 10);

$media->manipulations = [
    'resized' => $resizedConversion->toArray()
];

```

### CODE REVIEW

File: src/Manipulations.php [hunk 1]
```diff
@@ -59,7 +59,14 @@ class Manipulations
 
     public function __construct(array $manipulations = [])
     {
-        $this->manipulationSequence = new ManipulationSequence($manipulations);
+        //Check if is a single array or multi arrays
+        if ($this->checkIfHasMultipleConversions($manipulations)) {
+            foreach ($manipulations as $manipulation) {
+                $this->manipulationSequence = new ManipulationSequence($manipulation);
+            }
+        } else {
```
Comment (freekmurze) [inline src/Manipulations.php]: This `else` statement can be avoided by wrapping `$manipulations` in an extra array if it does not contain multiple conversions.

File: src/Manipulations.php [hunk 2]
```diff
@@ -555,6 +562,43 @@ public function apply()
...
+     *
+     * @param  Array $manipulations
+     * @return bool
+     */
+    private function checkIfHasMultipleConversions($manipulations)
+    {
+        $hasArrays = false;
+        foreach ($manipulations as $manipulation) {
+            if (isset($manipulation[0]) && is_array($manipulation[0])) {
+                $hasArrays = true;
```
Comment (freekmurze) [inline src/Manipulations.php]: use an early return here

File: src/Manipulations.php [hunk 3]
```diff
@@ -555,6 +562,43 @@ public function apply()
...
+
+    /**
+     * Checks if the given manipulations has arrays inside or not
+     *
+     * @param  Array $manipulations
+     * @return bool
+     */
+    private function checkIfHasMultipleConversions($manipulations)
+    {
+        $hasArrays = false;
```
Comment (freekmurze) [inline src/Manipulations.php]: this line can be removed

File: src/Manipulations.php [hunk 4]
```diff
@@ -555,6 +562,43 @@ public function apply()
...
+        foreach ($manipulations as $manipulation) {
+            if (isset($manipulation[0]) && is_array($manipulation[0])) {
+                $hasArrays = true;
+            }
+        }
+        return $hasArrays;
```
Comment (freekmurze) [inline src/Manipulations.php]: just return `false` here.

File: src/Manipulations.php [hunk 5]
```diff
@@ -555,6 +562,43 @@ public function apply()
...
+        return $this->manipulationSequence->toArray();
+    }
+
+    /**
+     * Checks if the given manipulations has arrays inside or not
+     *
+     * @param  Array $manipulations
+     * @return bool
+     */
+    private function checkIfHasMultipleConversions($manipulations)
```
Comment (freekmurze) [inline src/Manipulations.php]: rename to `hasMultipleConverions` and add an `array` typehint and `bool` return type hint

File: src/Manipulations.php [hunk 6]
```diff
@@ -555,6 +562,43 @@ public function apply()
...
+     * Return manipulationSequence as array
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        return $this->manipulationSequence->toArray();
+    }
+
+    /**
```
Comment (freekmurze) [inline src/Manipulations.php]: Docblock may be removed

File: src/Manipulations.php [hunk 7]
```diff
@@ -555,6 +562,43 @@ public function apply()
         return $this;
     }
 
+    /**
+     * Create new manipulations class
+     *
+     * @return self
+     */
+    public static function create()
```
Comment (freekmurze) [inline src/Manipulations.php]: Allow passing an array with manipulations here. Pass it to the contructor.

File: src/Manipulations.php (hunk 1 above)
Comment (Krato) [inline src/Manipulations.php]: Sorry, I don't understand.

Comment (Krato) [inline src/Manipulations.php]: Is like that?

```php
if (!$this->hasMultipleConversions($manipulations)) {
    $manipulations = [$manipulations];
}

foreach ($manipulations as $manipulation) {
    $this->manipulationSequence = new ManipulationSequence($manipulation);
}
```

**Comment by Krato** [comment]
Updated. How you like it now! 👍

File: src/Manipulations.php (hunk 6 above)
Comment (Krato) [inline src/Manipulations.php]: Removed

File: src/Manipulations.php [hunk 8]
```diff
@@ -59,7 +59,13 @@ class Manipulations
 
     public function __construct(array $manipulations = [])
     {
-        $this->manipulationSequence = new ManipulationSequence($manipulations);
+        if (!$this->hasMultipleConversions($manipulations)) {
```
Comment (freekmurze) [inline src/Manipulations.php]: Much better 👍

File: src/Manipulations.php [hunk 9]
```diff
@@ -555,6 +561,41 @@ public function apply()
...
+     *
+     * @return self
+     */
+    public static function create(array $manipulations = [])
+    {
+        return new self($manipulations);
+    }
+
+    
+    public function toArray()
```
Comment (freekmurze) [inline src/Manipulations.php]: Add an `array` return type hint here

**Comment by Krato** [comment]
Do it!

**Comment by Krato** [comment]
Is everything ok now yes?

**Comment by freekmurze** [comment]
Haven't come around to this one yet. Holiday 🌴

**Comment by Krato** [comment]
Enjoy 😄

**Comment by freekmurze** [comment]
Thank you!

**Comment by Krato** [comment]
Thanks for the merge 😄

**Pull request merged**